frontend/src/scenes/session-recordings/player/playerUtils.ts

```diff
@@ -0,0 +1,158 @@
+import { MouseEvent as ReactMouseEvent, TouchEvent as ReactTouchEvent } from 'react'
+import { PlayerPosition, RecordingSegment, RecordingStartAndEndTime } from '~/types'
+
+export const THUMB_SIZE = 15
+export const THUMB_OFFSET = THUMB_SIZE / 2
+
+export type ReactInteractEvent = ReactMouseEvent<HTMLDivElement, MouseEvent> | ReactTouchEvent<HTMLDivElement>
+export type InteractEvent = MouseEvent | TouchEvent
+
+export function isTouchEvent(
+    event: ReactInteractEvent | InteractEvent
+): event is ReactTouchEvent<HTMLDivElement> | TouchEvent {
+    return 'touches' in event
+}
+
+export function isMouseEvent(
+    event: ReactInteractEvent | InteractEvent
+): event is ReactMouseEvent<HTMLDivElement> | MouseEvent {
+    return 'clientX' in event
+}
+
+export const getXPos = (event: ReactInteractEvent | InteractEvent): number => {
+    if (isTouchEvent(event)) {
+        return event?.touches?.[0]?.pageX ?? event?.changedTouches?.[0]?.pageX // x coordinates are in changedTouches on touchend
+    }
+    if (isMouseEvent(event)) {
+        return event?.clientX
+    }
+    return 0
+}
+
+// Returns a positive number if a is greater than b, negative if b is greater than a, and 0 if they are equal
+export function comparePlayerPositions(a: PlayerPosition, b: PlayerPosition, segments: RecordingSegment[]): number {
+    if (a.windowId === b.windowId) {
+        return a.time - b.time
+    }
+    for (const segment of segments) {
+        if (
+            a.windowId === segment.windowId &&
+            a.time >= segment.startPlayerPosition.time &&
+            a.time <= segment.endPlayerPosition.time
+        ) {
+            return -1
+        } else if (
+            b.windowId === segment.windowId &&
+            b.time >= segment.startPlayerPosition.time &&
+            b.time <= segment.endPlayerPosition.time
+        ) {
+            return 1
+        }
+    }
+    throw `Could not find player positions in segments`
+}
+
+export function getSegmentFromPlayerPosition(
+    playerPosition: PlayerPosition,
+    segments: RecordingSegment[]
+): RecordingSegment | null {
+    for (const segment of segments) {
+        if (
+            playerPosition.windowId === segment.windowId &&
+            playerPosition.time >= segment.startPlayerPosition.time &&
+            playerPosition.time <= segment.endPlayerPosition.time
```

Comment:
Is there a chance that two segments will share an end time and start time (or start and end) respectively? If this happens, the player might fail to seek to the next segment because it can never leave the current segment. But this might be a non-issue if the backend guarantees no overlap.

Reply:
That makes sense, I made it to the .py files and see that you take care of this overlap pretty well. If you think there's an easy way to protect against this in the frontend it might be worth it, but as it is now seems okay to me.